src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -31785,6 +31782,70 @@ static void M(C c, string? p1, string? p2)
             CompileAndVerify(comp2);
         }
 
+        [Fact]
+        public void NotNullIfNotNull_ByValueParameter_02()
+        {
+            var source = """
+                using System.Diagnostics.CodeAnalysis;
+
+                class C
+                {
+                    public static void M(string? p, [NotNullIfNotNull("p")] string? p2) => throw null!;
+                }
+                class D
+                {
+                    static void M(string p, string? p2)
+                    {
+                        C.M(p, p2);
```

Comment:
>C.M(p, p2) [](http://example.com/codeflow?start=24&length=10)

What should be the behavior for the following?
```
C.M(p, p2, p2 = null);
p.ToString();
p2.ToString();
```

Assuming
```
public static void M(string? p, [NotNullIfNotNull("p")] string? p2, string? p3) {...}
```

Reply:
I think it shouldn't report a warning, for consistency with other existing scenarios. When a variable is read and subsequently written within an argument list, we sometimes don't react perfectly to the changes in state. For example: ([SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAAQDCAsAFADeVx9x+AzIwIwBsjALMQLIAUyAKYAzNgAZiAB1ZwJAfmmEAlMQC8APmmsAdABUA9gGUEyWAHN+ygNxUAvlSLEAIlVqUGbTvh69oMKzoGd09PADcIZCV1YgAiCFjbD1D6Uh0BYTEpQjlsmJhUABtCm2IAejLiGANiAHdImAsg+gdKOyA==))

```cs
class C
{
    public static void M([NotNull] string? p1, string? p2)
    {
        if (p1 is null) throw new Exception();
    }
}
class D
{
    static void Main()
    {
        var p2 = "a";
        C.M(p2, p2 = null);
        p2.ToString(); // no warning
    }
}
```